### ISSUE TITLE
Some questions about the implementation of Motion Frame.

### ISSUE THREAD

**Issue by zhysora** [fudan-generative-vision/hallo#99]
Thanks for your greats works! 
1. It seems that the "Motion Frame" only worked in UNetMidBlock3DCrossAttn. Because "self.gradient_checkpointing" is always False in CrossAttnDownBlock3D and CrossAttnUpBlock3D. (referred to https://github.com/fudan-generative-vision/hallo/blob/main/hallo/models/unet_3d_blocks.py#L1133).
2. Is there is a bug in (https://github.com/fudan-generative-vision/hallo/blob/main/hallo/models/unet_3d_blocks.py#L1133)? n_motion_frame is set 2 in the inference, But it is padded with 4 frames here.

**Comment by xumingw** [comment]
1. Regarding the gradient checkpointing, we apologize for any confusion. Gradient checkpointing is indeed enabled in CrossAttnDownBlock3D and CrossAttnUpBlock3D as well.
2. As for the snippet you referred to, it is a test part that we forgot to delete. In the final version, the motion frame padding is handled in the main inference entry, it doesn't go to the else branch.

**Comment by zhysora** [comment]
> 1. Regarding the gradient checkpointing, we apologize for any confusion. Gradient checkpointing is indeed enabled in CrossAttnDownBlock3D and CrossAttnUpBlock3D as well.
> 2. As for the snippet you referred to, it is a test part that we forgot to delete. In the final version, the motion frame padding is handled in the main inference entry, it doesn't go to the else branch.

Thanks for your reply. 
1. But the "self.training" is False in the inference. It seems to go to the else branch, which may disable the "Motion Frame", too. (referred to https://github.com/fudan-generative-vision/hallo/blob/main/hallo/models/unet_3d_blocks.py#L1133).

**Comment by xumingw** [comment]
The "self.training" is still True in inference. It caused some confusion, but doesn't affect inference. we may fix it in future

**Comment by zhysora** [comment]
> The "self.training" is still True in inference. It caused some confusion, but doesn't affect inference. we may fix it in future

Thanks for your reply and the released training scripts. There some new confusions:

1. In training the motion frame is padded with zeros (referred to https://github.com/fudan-generative-vision/hallo/blob/main/scripts/train_stage2.py#L804), but in inference it is padded with the reference images (referred to https://github.com/fudan-generative-vision/hallo/blob/main/scripts/train_stage2.py#L357). 
2. About the uncond forward, the reference net is disabled according to (https://github.com/fudan-generative-vision/hallo/blob/main/scripts/train_stage2.py#L148) when training. The features of motion frames may not be processed too, which seems to be not aligned with the inference codes.
